shiny/driver/internal/win32: restore normalize / minimize / maximze

200sc opened the pull request:
For titlebar support the desktop OSes should support these methods. 

- [x] WIndows support 
- [x] Linux support
- [x] OSX support 
- [x] Example (maybe in grove)
- [x] Passing tests

200sc committed:
shiny/driver/internal/win32: restore normalize / minimize / maximze

nfuden commented on shiny/driver/mtldriver/window.go:
```diff
@@ -211,6 +188,39 @@ func (w *Window) Publish() {
 	cb.Commit()
+}
+
+func (w *Window) GetDesktopPosition() (x, y float64) {
```
Ooh what a great piece of fxnality

200sc committed:
shiny/driver/mtldriver: add min/max/normalize implementation

200sc committed:
examples/screenopts: include min/max/normalize

200sc committed:
shiny/driver/x11driver: add min/max/normalize

minimize is unimplemented; it needs to either use XIconifyWindow which is unavailable in currect xproto libraries or send some kind of minimize message to the root window

200sc committed:
shiny/driver: correct go build directives

200sc committed:
window:  integrate now non-optional window manipulation methods

200sc committed:
.github/workflows: bump go version
